### PR TITLE
Office online viewer

### PR DESCRIPTION
**Pull request by davidkreidler**

### CODE REVIEW

**Comment by MasterCarl** [comment]
screen shot 2017-06-14 at 11 13 17
Doesn't show correctly in Safari

File: static/scripts/jquery.cookie.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+		return result;
+	};
+
+	config.defaults = {};
+
+	$.removeCookie = function (key, options) {
+		// Must not alter options, thus extending a fresh object...
+		$.cookie(key, '', $.extend({}, options, { expires: -1 }));
+		return !$.cookie(key);
```
Comment (MasterCarl) [inline static/scripts/jquery.cookie.js]: This seems strange

File: static/scripts/files.js [hunk 2]
```diff
@@ -265,55 +268,119 @@ $moveModal.modal('hide');
...
-            case 'image/'+filetype.substr(6, filetype.length) :
-                $('#picture').attr("src", '/files/file?file='+file);
-                break;
-            case 'audio/'+filetype.substr(6, filetype.length):
-            case 'video/'+filetype.substr(6, filetype.length):
-                videojs('my-video').ready(function () {
-                    this.src({type: filetype, src: '/files/file?file='+file});
+}
+
+function openInIframe(source){
```
Comment (MasterCarl) [inline static/scripts/files.js]: Maybe add a comment what this function does

File: static/scripts/files.js [hunk 3]
```diff
@@ -265,55 +268,119 @@ $moveModal.modal('hide');
...
+
+        case 'text/plain': //only in Google Docs Viewer                                     //.txt
+        case 'application/octet-stream':                                                    //.psd
+        case 'application/x-zip-compressed':                                                //.zip
+            var gviewer ="https://docs.google.com/viewer?url=";
+            var url = window.location.href;
+            url = url.substr(0, url.lastIndexOf("/"));
+            url = url.substr(0, url.lastIndexOf("/"));
+            url += "/files/file?file=" + file;
+            $openModal.find('.modal-title').text("Möchtest du diese Datei mit dem externen Dienst Google Docs Viewer ansehen?");
```
Comment (MasterCarl) [inline static/scripts/files.js]: I would pass this as a variable to `openInIframe`

File: static/scripts/files.js [hunk 4]
```diff
@@ -265,55 +268,119 @@ $moveModal.modal('hide');
...
+
+function videoClick(e) {
+    e.stopPropagation();
+    e.preventDefault();
+}
+
+function fileViewer(filetype, file) {
+    $('#my-video').css("display","none");
+    switch (filetype) {
+        case 'image/'+filetype.substr(6, filetype.length) :
```
Comment (MasterCarl) [inline static/scripts/files.js]: Could you extract `filetype.substr(6, filetype.length)` into a variable?

Comment (MasterCarl) [inline static/scripts/files.js]: `.substr(6)` should also work

File: static/scripts/jquery.cookie.js (hunk 1 above)
Comment (davidkreidler) [inline static/scripts/jquery.cookie.js]: copied from  jQuery Cookie Plugin v1.4.1

File: static/scripts/files.js (hunk 3 above)
Comment (davidkreidler) [inline static/scripts/files.js]: added complexity (new switch and variables). Not necessary for two viewers

**Pull request merged**